### ISSUE TITLE
[Bug/Feature Request] Safari support

### ISSUE THREAD

**Issue by Smexhy** [Lissy93/dashy#894]
### Question

Hello, I was looking through everywhere but couldn't find any reason why Safari isn't supported browser for Dashy. I would love to use it, but since our devices are mainly from apple ecosystem, we are using Safari a lot. I would like to know why if possible. Thank you.

### Category

Development

### Please tick the boxes

- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy (check the first two digits of the version number)
- [X] You've checked that this [question hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by Dima-Kal** [comment]
I don't experience any issues running on safari

**Comment by Smexhy** [comment]
> 

I have only issues with the minimalist style of the UI:

<img width="1440" alt="CleanShot 2022-09-11 at 23 46 11@2x" src="https://user-images.githubusercontent.com/4880625/189550233-7c0f2d4b-79c8-4ff1-be66-aba83b358aa9.png">

and some small issues with the default UI, but I was just curious more than anything. Also when clicking on a phone (on iphone and on safari) I am always opening contextual menu for editing, which can be turned off in settings, so it's not that big of a deal either, it's just those small little things :)

**Comment by Lissy93** [comment]
Yeah, so Safari has a few weird quirks, but nothing which can't be fixed. To be honest, the main reason it's not officially supported, is because I don't have an Apple device I can test it on. But if you can send me screenshots of any bugs, I should be able to fix them just by sight.

Is that one in the screenshot the only issue you've come across, or is there more?

**Comment by Smexhy** [comment]
> Yeah, so Safari has a few weird quirks, but nothing which can't be fixed. To be honest, the main reason it's not officially supported, is because I don't have an Apple device I can test it on. But if you can send me screenshots of any bugs, I should be able to fix them just by sight.
> 
> Is that one in the screenshot the only issue you've come across, or is there more?

I fully understand that, thank you for your answer. It is one of the worst problems I encountered, but not the only one for sure. I found quite a few while setting it up, but didn't take notes which is actually my bad. I will need to give it a look again. I would definitely took notes if I knew you would be willing to have a look at those, just wasn't sure since it's not even listed as supported browser and I know you have other stuff to fix and do for the supported browsers. Though I remember few of them:

There is a problem with the logo (if the user uploads one in settings). I was able to fix it with custom CSS by setting it a fixed height, otherwise it would look like this:

<img width="903" alt="CleanShot 2022-09-12 at 00 35 23@2x" src="https://user-images.githubusercontent.com/4880625/189551736-1a48d491-e6fb-402b-8dd2-3f51c61ebed8.png">

Next issue I noticed is that if I use custom CSS option in settings (the one that is used locally to browser, not globally) it doesn't actually work and messes some other stuff on the page. The only way to fix this is by removing local storage for that specific webpage, which reverts all the CSS changes with it.

Another stuff from what I recall is the issue with widgets on safari - or at least with few of them, in my case with glances. I checked on google chrome on my second device that I use, and it was correct there, so this has to be yet again because of safari. The percentage value for the current CPU and memory usage is not in the middle - which is causing the value to be almost invisible, depending on the color of the section and theme, in my case yellow. The value is put into the graph instead of the middle. This can be also simply fixed by the custom CSS, which I didn't have chance to do so yet.

<img width="772" alt="CleanShot 2022-09-12 at 00 40 09@2x" src="https://user-images.githubusercontent.com/4880625/189551940-24f08285-1196-430b-b93a-be88b62437ac.png">

Lastly, what I already mentioned in the previous reply here, while using the Safari app on my iPhone, while navigating and scrolling through the website UI or while opening links, I mostly always open the built-in contextual menu to edit stuff. I am not sure if this is intended or not since I don't have other device to try it out, and yes, we do have option to turn this contextual menu off like I mentioned as well, but what I personally would love to be able to is to disable it only on phones (since it can be pretty useful on desktop), or perhaps change how the right click is registered on the phones if somehow possible, since on smartphone we don't have option of difference between left / right click like on desktop, but just touch.

And now to deviate a bit from the main topic, I would also love to have some status of the rebuilding happening after making changes to the configuration from the UI. Often after making changes I am not sure if my changes were wrong or if it's cache, or if it's still rebuilding and without looking directly in logs it's a bit problematic to find out.

Otherwise, to not only make this look like I am searching for every possible bug or error in design etc. I am very happy altogether with Dashy, since it looks really classy, simple and minimalistic, works fast and is really one of a kind. One of the best home-fronts for my server I ever had. Changed from Organizr and I am happy I finally did, since it's much better looking and quick in my opinion.

**Comment by CrazyWolf13** [comment]
@Lissy93 Think this is still relevant

**Comment by CrazyWolf13** [comment]
> > Yeah, so Safari has a few weird quirks, but nothing which can't be fixed. To be honest, the main reason it's not officially supported, is because I don't have an Apple device I can test it on. But if you can send me screenshots of any bugs, I should be able to fix them just by sight.
> > Is that one in the screenshot the only issue you've come across, or is there more?
> 
> I fully understand that, thank you for your answer. It is one of the worst problems I encountered, but not the only one for sure. I found quite a few while setting it up, but didn't take notes which is actually my bad. I will need to give it a look again. I would definitely took notes if I knew you would be willing to have a look at those, just wasn't sure since it's not even listed as supported browser and I know you have other stuff to fix and do for the supported browsers. Though I remember few of them:
> 
> There is a problem with the logo (if the user uploads one in settings). I was able to fix it with custom CSS by setting it a fixed height, otherwise it would look like this:
> 
> <img alt="CleanShot 2022-09-12 at 00 35 23@2x" width="903" src="https://user-images.githubusercontent.com/4880625/189551736-1a48d491-e6fb-402b-8dd2-3f51c61ebed8.png">
> 
> Next issue I noticed is that if I use custom CSS option in settings (the one that is used locally to browser, not globally) it doesn't actually work and messes some other stuff on the page. The only way to fix this is by removing local storage for that specific webpage, which reverts all the CSS changes with it.
> 
> Another stuff from what I recall is the issue with widgets on safari - or at least with few of them, in my case with glances. I checked on google chrome on my second device that I use, and it was correct there, so this has to be yet again because of safari. The percentage value for the current CPU and memory usage is not in the middle - which is causing the value to be almost invisible, depending on the color of the section and theme, in my case yellow. The value is put into the graph instead of the middle. This can be also simply fixed by the custom CSS, which I didn't have chance to do so yet.
> 
> <img alt="CleanShot 2022-09-12 at 00 40 09@2x" width="772" src="https://user-images.githubusercontent.com/4880625/189551940-24f08285-1196-430b-b93a-be88b62437ac.png">
> 
> Lastly, what I already mentioned in the previous reply here, while using the Safari app on my iPhone, while navigating and scrolling through the website UI or while opening links, I mostly always open the built-in contextual menu to edit stuff. I am not sure if this is intended or not since I don't have other device to try it out, and yes, we do have option to turn this contextual menu off like I mentioned as well, but what I personally would love to be able to is to disable it only on phones (since it can be pretty useful on desktop), or perhaps change how the right click is registered on the phones if somehow possible, since on smartphone we don't have option of difference between left / right click like on desktop, but just touch.
> 
> And now to deviate a bit from the main topic, I would also love to have some status of the rebuilding happening after making changes to the configuration from the UI. Often after making changes I am not sure if my changes were wrong or if it's cache, or if it's still rebuilding and without looking directly in logs it's a bit problematic to find out.
> 
> Otherwise, to not only make this look like I am searching for every possible bug or error in design etc. I am very happy altogether with Dashy, since it looks really classy, simple and minimalistic, works fast and is really one of a kind. One of the best home-fronts for my server I ever had. Changed from Organizr and I am happy I finally did, since it's much better looking and quick in my opinion.

How did you solve it with custom css?

**Comment by CrazyWolf13** [comment]
#1358  seems also relevant here.

**Comment by Juicytronic** [comment]
Would be great if this would happen.
Dashy is absolute unusable on my iPhone 14 Pro.
The dashboard doesn't even load, neither with Safari, Chrome, Opera or Firefox.

**Comment by CrazyWolf13** [comment]
Also linking #1443 to this.

**Comment by bobbyl140** [comment]
> Also linking #1443 to this.

Thank you!

**Comment by bobbyl140** [comment]
> I have only issues with the minimalist style of the UI:
> 
> <img alt="CleanShot 2022-09-11 at 23 46 11@2x" width="1440" src="https://user-images.githubusercontent.com/4880625/189550233-7c0f2d4b-79c8-4ff1-be66-aba83b358aa9.png">

I realize this comment is over a year old, but I can also report the same issue.

**Comment by lissy93** [comment]
Closing as resolved, as Safari 26.0.0 does now support all CSS properties used in Dashy.
